schemas/io-package.json

```diff
@@ -0,0 +1,457 @@
+                    "type": "boolean"
+                },
+                "main": {
+                    "description": "[DEPRECATED] Location of adapter main file",
+                    "type": "string"
+                },
+                "materialize": {
+                    "description": "If adapter supports > admin3 (materialize style)",
+                    "type": "boolean"
+                },
+                "materializeTab": {
+                    "description": "If adapter supports > admin3 for tab (materialize style)",
+                    "type": "boolean"
+                },
+                "mode": {
+                    "description": "Execution mode of the adapter",
+                    "type": "string",
+                    "enum": ["none", "daemon", "subscribe", "schedule", "once", "extension"]
+                },
+                "noConfig": {
+                    "description": "Do not show configuration dialog for instance",
+                    "type": "boolean"
+                },
+                "noIntro": {
+                    "description": "never show instances of this adapter on Intro/Overview screen in admin (like icons, widgets)",
+                    "type": "boolean"
+                },
+                "noRepository": {
+                    "description": "If adapter delivered with initial installation or has own repository",
+                    "type": "boolean"
+                },
+                "nogit": {
+                    "description": "If true, no install from github directly is possible",
+                    "type": "boolean"
+                },
+                "nondeletable": {
+                    "description": "This adapter cannot be deleted or updated. It will be updated together with controller.",
+                    "type": "boolean"
+                },
+                "onlyWWW": {
+                    "description": "Say to controller, that adapter has only html files and no main.js, like rickshaw",
+                    "type": "boolean"
+                },
+                "osDependencies": {
+                    "description": "OS packages which should be installed on adapter installation",
+                    "type": "object",
+                    "additionalProperties": false,
+                    "patternProperties": {
+                        "linux|darwin|win32": {
+                            "type": "array",
+                            "items": {
+                                "type": "string"
+                            }
+                        }
+                    }
+                },
+                "os": {
+                    "description": "String or array of supported operation systems, e.g [\"linux\", \"darwin\"]",
+                    "type": ["string", "array"],
+                    "minItems": 1,
+                    "items": {
+                        "type": "string"
+                    }
```

Comment:
We can be more specific here:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/18bbb658a29f954590be5f0970f8c94e943f296e/types/iobroker/objects.d.ts#L446

And I'm not sure if that definition works. I think you need `"anyOf"`

Reply:
Thanks. Of course it's the same pattern as a few liens above for "osDependencies". 

The definition works, I've tested it.